numpy/_core/arrayprint.py

```diff
@@ -1443,10 +1472,11 @@ def _void_scalar_to_string(x, is_repr=True):
     scalartypes.c.src code, and is placed here because it uses the elementwise
     formatters defined above.
     """
-    options = _format_options.copy()
+    options = _format_options.get().copy()
```

Comment:
Not sure it makes sense, but I was half hoping we can just stop passing options around completely and enter a new context ourselves when necessary.  `contextvar` lookup is almost as fast as dict lookup, and global lookup also not really expensive at the function level (within a loop it might be).